### ISSUE TITLE
Feedback typemill version 2

### ISSUE THREAD

**Issue by trendschau** [trendschau/typemill#407]
I published a developer version of typemill 2. 

!!! DO NOT USE IT IN PRODUCTION !!!

But you can install it locally and have a first look. It would be very helpful to get some general feedback or some specific feedback about missing features or bugs.

You can use the git-version in the branch tm2-dev or the ZIP-version published here: https://typemill.net/news/typemill-v2-dev-release

**Comment by maciejtwardy** [comment]
Hi,

Two errors after the first quick launch of Typemill v2:
1. When trying to delete first level folders, a short browser message "[object Object]" appears and the folder is not deleted.

2. After adding a sample simple content going to view (Frontend) I get an error:

Warning: Attempt to read property "keyPathArray" on bool in /var/www/typemill/system/typemill/Controllers/ControllerWebFrontend.php on line 79
Warning: Attempt to read property "pathWithoutType" on bool in /var/www/typemill/system/typemill/Models/Navigation.php on line 522
Warning: Attempt to read property "keyPathArray" on bool in /var/www/typemill/system/typemill/Models/Navigation.php on line 527

Slim Application Error
The application could not run because of the following error:

Details
Type: TypeError
Code: 0
Message: count(): Argument #1 ($value) must be of type Countable|array, null given
File: /var/www/typemill/system/typemill/Models/Navigation.php
Line: 527

**Comment by trendschau** [comment]
Thank you for reporting! 

I fixed the first error and hopefully also the second (could not reproduce) in release candidate 1 that I just published here: https://typemill.net/news/typemill-v2-rc1

**Comment by sopkomir** [comment]
Hi, error in the blog page (from posts folder) - link: https://tm.obciankari.sk/blog

`Warning: Undefined array key "title" in /data/c/2/c296cd35-9cab-4966-9e8d-e82b5760b056/obciankari.sk/sub/tm/system/typemill/Extensions/TwigMetaExtension.php on line 25`

Miro Sopko
www.eduvolucia.sk

**Comment by trendschau** [comment]
thank you for reporting, will fix with next release candidate

**Comment by Alzarath** [comment]
Disclosures: I'm using lighttpd which as I understand it is unsupported. I'm using PHP 8.2.7. I've made file changes to fix many of the following issues on my end to get it in working order.

There are a couple errors along the lines of `Unparenthesized a ? b : c ? d : e is not supported` that prevent use.

There are a lot of Deprecation reports related to the "Creation of dynamic property".

CSS for the navigation on the left side of the admin pages does not indent pages beyond `.pl-12`. The classes `.pl-15`, `.pl-18`, etc are stuck to the left which can make navigation a bit confusing.

Most of the meta information—especially the Navigation Title, Meta Title, and Meta Description—appear to reflect the top-most folder. In my case the "home" folder. Changing the values has thus far modified the correct entry's information, but future visits will continue to show information for Home. The `slug` does appear to reflect appropriately, however.

There is no "Refresh Cache" button like there was in previous versions, only the automatic cache refresh, which is inconvenient when I make frequent changes to files and want to update immediately.

**Comment by trendschau** [comment]
Thank you for the feedback, Sorry for that, I changed a method name for the final bugfix and that produced a lot of mess.

I will check everything again and also test for 8.2.11 now. 

The refresh cache feature will be added later, because it will be part of an additional user interface that will be launched later...

**Comment by robs-fotoalbum** [comment]
I tried rc2 and encountered the following:

1. During setup, I had to add a folder "data" which was expected - after that was done, setup was possible
2. Had some issues with an automatic generated password in Firefox - obviously not all characters were allowed. A simpler password was fine.
3. The hero image that I wanted to use from the homepage in my landing page was not displayed. It was uploaded but to a temp folder. Moving it to the media folder didn't solve the issue. There seems to be some minor code adjustment necessary.

I like the new and fresh look and the streamlined workflow of Typemill 2. Looking forward to the final version and migrating my site.
